packages/form-builder/src/containers/HeaderContainer.tsx

```diff
@@ -85,12 +88,25 @@ const HeaderContainer = ({
             contentRight={
                 <>
                     <Button
-                        onClick={onSave}
+                        onClick={() => onSave('draft')}
+                        aria-disabled={isSaving || !isDirty}
+                        disabled={isSaving || !isDirty}
+                        variant="tertiary"
+                    >
+                        {'draft' === formSettings.formStatus ? __('Save as Draft', 'give') : __('Switch to Draft', 'give')}
+                    </Button>
+                    <Button
+                        onClick={() => onSave('publish')}
                         aria-disabled={isSaving || !isDirty}
                         disabled={isSaving || !isDirty}
                         variant="primary"
                     >
-                        {isSaving ? __('Updating...', 'give') : __('Update', 'give')}
+                        {isSaving
+                            ? __('Updating...', 'give')
+                            : 'publish' === formSettings.formStatus
+                                ? __('Update', 'give')
+                                : __('Publish', 'give')
+                        }
```

Comment:
I just want to point out a few things here that I noticed.

1. When you first create the form, it's in draft but the publish button is disabled.  We need to make sure publish can _always_ be clicked when in draft mode (regardless if dirty) so people can always publish the form when satisfied.  Then we can go back to checking dirty state if published on the update button. 
2. Clicking "save as draft" should modify the test to `saving...` but not the modify the publish button text like below:

<img width="306" alt="Screenshot 2023-03-31 at 10 25 12 AM" src="https://user-images.githubusercontent.com/10138447/229149392-2bcee2cb-8047-4d3e-9c5b-47bfa6abfd42.png">

3. When a form is in draft mode - we should be able to view the form on the front-end when logged in (That's how native WP posts/post-types work by using the postID instead of the slug)

Reply:
I updated the draft and publish button text and disabled states to account for the different circumstances, see https://github.com/impress-org/givewp-next-gen/pull/147/commits/bf00d2a34574a719a151ad844ea2f92383fbedf8.

Reply:
I confirmed that draft forms can be viewed at the permalink when authenticated - and that draft forms have a 404 redirect when unauthenticated.